packages/mobile/src/consumerIncentives/ConsumerIncentivesHomeScreen.tsx

```diff
@@ -140,6 +152,69 @@ function SuperchargingInfo() {
   )
 }
 
+function ClaimSuperchargeRewards({ rewards }: { rewards: SuperchargePendingReward[] }) {
+  const tokens = useSelector(tokensByAddressSelector)
+  const { t } = useTranslation()
+
+  const rewardsByToken: { [tokenAddress: string]: BigNumber | undefined } = rewards.reduce(
+    (acc, reward) => {
+      const tokenAddress = reward.tokenAddress.toLowerCase()
+      if (!acc[tokenAddress]) {
+        acc[tokenAddress] = new BigNumber(0)
+      }
+      acc[tokenAddress] = acc[tokenAddress].plus(
+        new BigNumber(reward.amount, 16).div(WEI_PER_TOKEN)
+      )
+      return acc
+    },
+    {} as { [tokenAddress: string]: BigNumber }
+  )
+
+  const rewardStrings = Object.entries(rewardsByToken).map(
+    ([token, amount]) => `${amount?.toFixed(2)} ${tokens[token]?.symbol}`
+  )
+  const [singleRewardToken, singleRewardAmount] = Object.entries(rewardsByToken)[0]
+
+  return (
+    <>
+      <Text style={[styles.title, { color: colors.greenUI }]} testID="ClaimSuperchargeDescription">
+        {rewardStrings.length > 1
+          ? t('superchargeRewardsAvailableMultipleTokens', { amount: rewardStrings.join(' & ') })
+          : t('superchargeRewardsAvailable', {
+              token: tokens[singleRewardToken]?.symbol,
+              amount: singleRewardAmount?.toFixed(2),
+            })}
+      </Text>
+      <Text style={styles.description}>{t('superchargeClaimText')}</Text>
+    </>
+  )
+}
+
+function useFetchAvailableRewards(): {
+  availableRewards: SuperchargePendingReward[]
+  loading: boolean
+} {
+  const address = useSelector(walletAddressSelector)
+
+  const { result, loading } = useAsync(async () => {
+    try {
+      const response = await fetch(`${SUPERCHARGE_AVAILABLE_REWARDS_URL}?address=${address}`, {
+        method: 'GET',
+        headers: {
+          'Content-Type': 'application/json',
+          Accept: 'application/json',
+        },
+      })
+
+      const responseBody = await response.json()
```

Comment:
nit: I'm wondering how should we handle the responses with status 400 and 500.
I think we are ignoring those error right now, and the user will think that there is no reward to be claimed.